models.py

```diff
@@ -588,6 +588,60 @@ def random_type(row):
     return random.choice(form_to_btype[form])
 
 
+def probable_type(row):
+    """
+    Function to pass to form_to_btype_callback in parcels_utils.add_buildings.
+    Reads form for the given row in new buildings DataFrame, gets the
+    building type probabilities from the form_btype_distributions injectable,
+    and chooses a building type based on those probabilities.
+
+    Parameters
+    ----------
+    row : Series
+
+    Returns
+    -------
+    btype : int
+    """
+    form = row['form']
+    form_to_btype_dists = orca.get_injectable("form_btype_distributions")
+    btype_dists = form_to_btype_dists[form]
+    # keys() and values() guaranteed to be in same order
+    btype = np.random.choice(a=btype_dists.keys(), p=btype_dists.values())
```

Comment:
If btype_dists was a Series we could use [`btype_dists.sample(n=1, weights=btype_dists).index`](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.sample.html?highlight=sample#pandas.Series.sample). But having typed it out I don't know if it is better.

Reply:
Either approach would work - I used dictionaries here instead of Series because I think they're just simpler to work with.

Reply:
Ok, dictionaries look good then. :-)